### PR TITLE
feat: drilldown arrow navigation

### PR DESCRIPTION
**Pull request by shawnyama**
# Description
* Upstream and downstream drilldowns can be navigated with chevron buttons
* Keyboard shortcuts work as well with `Shift` + left/right arrow keys
* scss changes prevent button from looking transparent when clicked, this whole time we were relying on the background to be white for these outline buttons

https://github.com/user-attachments/assets/da6b3dc4-9659-4b10-9af7-99d5f02ac260

### CODE REVIEW

File: packages/client/hmi-client/src/components/drilldown/tera-output-dropdown.vue
Comment (shawnyama) [inline packages/client/hmi-client/src/components/drilldown/tera-output-dropdown.vue:1]: Sorry about this unrelated fix, it just makes the output dropdown take up one line 
image

File: packages/client/hmi-client/src/components/drilldown/tera-drilldown.vue [hunk 1]
```diff
@@ -1,6 +1,28 @@
...
+			<Menu ref="upstreamMenu" class="ml-5" popup :model="upstreamOperatorsNav" :pt="menuPt" />
+			<template #tooltip-content>
+				<span class="operator-nav-info">
+					<span v-if="upstreamOperatorsNav?.[0]?.items?.length === 1">
+						<i :class="upstreamOperatorsNav[0].items[0].icon" />
+						<label>{{ upstreamOperatorsNav[0].items[0].label }}</label>
+					</span>
+					<label v-else>Upstream operators</label>
+					<span>
+						<kbd>Shift</kbd> + <kbd><i class="pi pi-arrow-left" /></kbd>
```
Comment (YohannParis) [inline packages/client/hmi-client/src/components/drilldown/tera-drilldown.vue:20]: Nice use of `<kbd>` 🎉

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/services/workflow.ts [hunk 2]
```diff
@@ -290,6 +290,51 @@ export const updateNodeStatus = (wf: Workflow, nodeId: string, status: OperatorS
...
+	const inputEdges = wf.edges.filter((edge) => edge.target === id);
+	const outputEdges = wf.edges.filter((edge) => edge.source === id);
+	const upstreamNodes = inputEdges
+		.map((edge) => wf.nodes.find((n) => n.id === edge.source))
+		.filter(Boolean) as WorkflowNode<any>[];
+	const downstreamNodes = outputEdges
+		.map((edge) => wf.nodes.find((n) => n.id === edge.target))
+		.filter(Boolean) as WorkflowNode<any>[];
+	return { upstreamNodes, downstreamNodes };
+};
```
Comment (mwdchang) [inline packages/client/hmi-client/src/services/workflow.ts:303]: You should build some kind of node-cache (id => node) to avoid unnecessary `O(n^2)` lookups. 

Also, consider adding in a unit test case for this.

Also, now you need to account for tombstone logic.

File: packages/client/hmi-client/src/components/workflow/tera-workflow.vue [hunk 3]
```diff
@@ -370,7 +376,8 @@ function updateOutputPort(node: WorkflowNode<any> | null, workflowOutput: Workfl
 }
 
 // Route is mutated then watcher is triggered to open or close the drilldown
-function addOperatorToRoute(nodeId: string | null) {
+function addOperatorToRoute(nodeId: string | null, animation: 'left' | 'right' | 'scale' = 'scale') {
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/workflow/tera-workflow.vue:379]: A small comment on what the animation parameter does would be nice

File: packages/client/hmi-client/src/services/workflow.ts [hunk 4]
```diff
@@ -290,14 +290,23 @@ export const updateNodeStatus = (wf: Workflow, nodeId: string, status: OperatorS
...
+	const findNeighborNode = (neighborId?: string) => {
+		if (!neighborId) return null;
+		if (!neighborNodeCache.has(neighborId)) {
+			const node = wf.nodes.find((n) => n.id === neighborId);
+			if (!node) return null;
+			neighborNodeCache.set(neighborId, node);
+		}
+		return neighborNodeCache.get(neighborId) ?? null;
+	};
+	console.log(neighborNodeCache);
```
Comment (YohannParis) [inline packages/client/hmi-client/src/services/workflow.ts]:
```suggestion
```

**Review by mwdchang — APPROVED** [review]
